### PR TITLE
Early check for invalid partition keys on launching a partition backfill

### PR DESCRIPTION
**Pull request by marijncv**
## Summary & Motivation

This adds an early check for on the partition keys that are supplied during a backfill mutation call. If an invalid partition key is found, a `PartitionKeysNotFoundError` is returned and the backfill is not scheduled for execution.

Contributes to https://github.com/dagster-io/dagster/issues/23801

## How I Tested These Changes

Unit tests

## Changelog

`LaunchPartitionBackfill` mutation now returns an early `PartitionKeysNotFoundError` if invalid partition keys are supplied.

- [x] `NEW` _(added new feature or capability)_
- [ ] `BUGFIX` _(fixed a bug)_
- [ ] `DOCS` _(added or updated documentation)_

### CODE REVIEW

File: python_modules/dagster-graphql/dagster_graphql/implementation/utils.py [hunk 1]
```diff
@@ -131,6 +135,58 @@ def assert_permission_for_asset_graph(
...
+    for asset_key, partition_subset in partition_subset_by_asset_key.items():
+        partitions_def = asset_graph.get(asset_key).partitions_def
+
+        if not partitions_def:
+            continue
+
+        for partition_key in partition_subset.get_partition_keys():
+            if not partitions_def.has_partition_key(
+                partition_key, backfill_datetime, dynamic_partitions_store
+            ):
```
Comment (smackesey) [inline python_modules/dagster-graphql/dagster_graphql/implementation/utils.py:186]: [1]

**Review by smackesey — CHANGES_REQUESTED** [review]
This seems useful. I made a comment about implementation but if you can address that and this passes tests, we can get this merged.

File: python_modules/dagster-graphql/dagster_graphql/implementation/utils.py [hunk 2]
```diff
@@ -131,6 +135,58 @@ def assert_permission_for_asset_graph(
...
+
+    partition_names = backfill.get_partition_names(graphene_info.context)
+
+    if not partition_names:
+        return
+
+    for partition_name in partition_names:
+        if not partitions_def.has_partition_key(
+            partition_name, backfill_datetime, dynamic_partitions_store
+        ):
```
[the remark this answers is not in the document]
Comment (marijncv) [reply python_modules/dagster-graphql/dagster_graphql/implementation/utils.py]: Great suggestion, thanks!!

**Review by smackesey — APPROVED** [review]
Looks good, thanks for this solid contribution!

**Pull request merged**